pgrx-tests/tests/ui/escaping-spiclient-1209-cursor.rs

```diff
@@ -0,0 +1,27 @@
+use pgrx::prelude::*;
+use std::error::Error;
+
+#[pg_test]
+fn issue1209() -> Result<Option<String>, Box<dyn Error>> {
+    // create the cursor we actually care about
+    let mut res = Spi::connect(|c| {
+        c.open_cursor("select 'hello' from generate_series(1, 10000)", None)
+            .fetch(10000)
+            .unwrap()
+    });
+
+    // here we just perform some allocations to make sure that the previous cursor gets invalidated
+    for _ in 0..1000 {
+        Spi::connect(|c| c.open_cursor("select 1", None).fetch(1).unwrap());
+    }
+
+    // later elements are probably more likely to point to deallocated memory
+    for _ in 0..1000 {
+        res.next();
+    }
+
+    // segfault
+    Ok(res.next().unwrap().get::<String>(1)?)
+}
+
+fn main() {}
```

Comment:
I don't like the frivolous mains but whatever.

Reply:
might be able to avoid with `#![crate_type = "lib"]`? not sure.

Reply:
Tried, unfortunately didn't work.